cps/tasks/download.py

```diff
@@ -41,44 +42,42 @@ def run(self, worker_thread):
                 p = process_open(subprocess_args, newlines=True)
 
                 # Define the pattern for the subprocess output
-                pattern_analyze = r"Running ANALYZE"
-                pattern_download = r"'action': 'download'"
+                pattern_progress = r"downloading"
 
                 while p.poll() is None:
                     line = p.stdout.readline()
                     if line:
-                        log.info(line)
-                        if pattern_analyze in line:
-                            log.info("Matched output (ANALYZE): %s", line)
-                            self.progress = 0.1
-                        if pattern_download in line:
-                            log.info("Matched output (download): %s", line)
-                            self.progress = 0.5
-                            
-                        p.wait()
-
-                        # Database operations
-                        requested_files = []
-                        conn = sqlite3.connect(SURVEY_DB_FILE)
-                        c = conn.cursor()
-                        c.execute("SELECT path FROM media")
-                        for row in c.fetchall():
-                            requested_files.append(row[0])
-
-                        c.execute("SELECT name FROM sqlite_master WHERE type='table' AND name='playlists'")
-                        if c.fetchone():
-                            c.execute("SELECT title FROM playlists")
-                            shelf_title = c.fetchone()[0]
-                        else:
-                            shelf_title = None
-                        conn.close()
-
-                        if self.original_url:
-                            response = requests.get(self.original_url, params={"requested_files": requested_files, "current_user_name": self.current_user_name, "shelf_title": shelf_title})
-                            if response.status_code == 200:
-                                log.info("Successfully sent the list of requested files to %s", self.original_url)
+                        if pattern_progress in line:
+                            percentage = int(re.search(r'\d+', line).group())
+                            if percentage < 100:
+                                self.progress = percentage / 100
                             else:
-                                log.error("Failed to send the list of requested files to %s", self.original_url)
+                                self.progress = 0.99
```

Comment:
FYI, I record the first 100% progress cycle as %99 so the script won't stop before yt-dlp finishes. There is often one or more quick 0%-100% cycles after the first one (for example the thumbnail and/or subtitles downloads)

Reply:
I definitely don't understand it yet, ah well 😎 

But thanks so much for explaining!

Reply:
> FYI, I record the first 100% progress cycle as %99 so the script won't stop before yt-dlp finishes. There is often one or more quick 0%-100% cycles after the first one (for example the thumbnail and/or subtitles downloads)

@deldesir ideally put some-or-all of this explanation in-line as a comment, near the Python code in question?

Reply:
Please put this explanation (or similar) into a comment near the actual Python code.

Then let's move forward with merging this PR.

Thanks!